### PR TITLE
Add torch.pad via identity broadcast and overwrite

### PR DESCRIPTION
**Pull request by GOavi101**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Implements `torch.nn.functional.pad` (constant padding) for the Spyre backend
using the identity broadcast + overwrite pattern introduced by `torch.cat`.

The decomposition of `aten.constant_pad_nd` proceeds in two steps:
1. **Fill**: create a 1-element scalar via `spyre::full`, broadcast-expand to
   the output shape, and materialize via `.clone()` (SFP identity broadcast).
   This zero-fills (or value-fills) the entire padded output.
2. **Copy**: call `spyre::overwrite` on the innermost padded dimension with the
   appropriate left offset. The `gap` between input and output size on that
   dimension maps to `backGap` in the SFP SDSC, causing the accelerator to
   skip the padded region automatically. Outer-dimension padding is already
   handled by the fill in step 1.

#### Special notes for your reviewer:

The approach mirrors `decompose_cat` in `decompositions.py`. The critical
insight is that `overwrite` only needs to be called for the innermost padded
dimension; outer-dimension padding rows are naturally left at the fill value
since `inner_fn` iterates over the input's shape only.

Test cases cover:
- 2D right-only padding on the last dim
- 2D padding on both last and outer dim simultaneously
- 3D right-only padding on the last dim
- 3D right-only padding on a non-last dim (outer dim only)

#### Does this PR introduce a user-facing change?

Yes 

#### Additional note:

### CODE REVIEW

File: torch_spyre/_inductor/decompositions.py [hunk 1]
```diff
@@ -603,6 +603,45 @@ def decompose_cat(
...
+    current = input
+    for i in range(n_dims_padded - 1, -1, -1):
+        left = pad[2 * i]
+        right = pad[2 * i + 1]
+        if left + right == 0:
+            continue
+        dim = input.dim() - 1 - i
+        intermediate_size = list(current.size())
+        intermediate_size[dim] += left + right
+        scalar = torch.ops.spyre.full([1], value, input.device, dtype=input.dtype)
```
Comment (avery-blanchard) [inline torch_spyre/_inductor/decompositions.py]: Is there a reason that the identity broadcast step here with the pad value value needs to be done in a loop?

Comment (GOavi101) [reply torch_spyre/_inductor/decompositions.py]: The scalar creation can be lifted out of the loop since it doesn't change between iterations. The intermediate tensor needs to be created per iteration because its size changes, but the scalar is constant. Will fix.

**Comment by michihirohorie** [comment]
@marnold-ibm @avery-blanchard @GOavi101 
The issue seems to be a requirement of processing at a granularity smaller than a stick on the stick dimension. This is not specific to padding. The same problem should occur in `torch.cat` along the stick dimension as well, so it would be preferable to discuss this as a separate op-common issue.

File: torch_spyre/_inductor/decompositions.py [hunk 2]
```diff
@@ -595,14 +595,81 @@ def decompose_cat(
...
+        raise Unsupported(
+            f"constant_pad_nd: left-padding is not supported on Spyre (pad={pad})"
+        )
+
+    # When the input's last dimension is smaller than one stick (64 fp16 elements),
+    # the data is repeated within each stick (inner_repeats = 64 // last_dim).
+    # The DSC requires the gap (right-pad on last dim) to be a multiple of
+    # inner_repeats; otherwise the hardware aborts with "Gap must be a multiple
+    # of inner repeats in expanded layout".
+    if n_dims_padded >= 1:
```
Comment (marnold-ibm) [inline torch_spyre/_inductor/decompositions.py]: I would vote that this check shouldn't be in this source code.   It's an orthogonal issue so this this limitation can be marked either with failing test cases (which I think is already being done above?) or an assertion / Unsupported check.

Also - n_dims isn't the only way this error occurs right?  A single dim with > 64 fails, so this check is not catching all the cases anyway.

**Comment by michihirohorie** [comment]
Looks good to me!

**Review by michihirohorie — APPROVED** [review]

**Review by michihirohorie — APPROVED** [review]

**Review by marnold-ibm — APPROVED** [review]
lgtm!

**Review by dgrove-oss — APPROVED** [review]
lgtm
